src/us/kbase/workspace/test/workspace/WorkspaceTest.java

```diff
@@ -6593,67 +6668,67 @@ public void listObjectsLimit() throws Exception {
+		final List<ObjectIdentifier> loi = new LinkedList<>();
+		final ObjectIdentifier.Builder b = ObjectIdentifier.getBuilder(wsi2);
+		loi.add(b.withID(5L).build());
+		loi.add(b.withID(15L).build());
 		ws.setObjectsDeleted(user, loi, true);
 		
 		loi.clear();
-		loi.add(new ObjectIdentifier(wsi, 7));
-		loi.add(new ObjectIdentifier(wsi, 17));
+		loi.add(b.withID(7L).build());
+		loi.add(b.withID(17L).build());
 		ws.setObjectsHidden(user, loi, true);
 		
 		objs.clear();
 		for (int i = 21; i < 31; i++) {
 			objs.add(new WorkspaceSaveObject(getRandomName(), new HashMap<String, String>(),
 					SAFE_TYPE1, null, new Provenance(user), false));
 		}
-		ws.saveObjects(user, wsi, objs, getIdFactory(0));
+		ws.saveObjects(user, wsi2, objs, getIdFactory(0));
 		
 		//test object pagination with deleted and hidden objects
-		checkObjectLimit(user, wsi, 5, 1, 6, nums(5));
-		checkObjectLimit(user, wsi, 6, 1, 8, nums(5, 7));
-		checkObjectLimit(user, wsi, 12, 1, 14, nums(5, 7));
-		checkObjectLimit(user, wsi, 13, 1, 16, nums(5, 7, 15));
-		checkObjectLimit(user, wsi, 14, 1, 18, nums(5, 7, 15, 17));
-		checkObjectLimit(user, wsi, 14, 1, 18, nums(5, 7, 15, 17));
-		checkObjectLimit(user, wsi, 15, 1, 19, nums(5, 7, 15, 17));
-		checkObjectLimit(user, wsi, 16, 1, 20, nums(5, 7, 15, 17));
-		checkObjectLimit(user, wsi, 17, 1, 21, nums(5, 7, 15, 17));
-		checkObjectLimit(user, wsi, 18, 1, 22, nums(5, 7, 15, 17));
-		checkObjectLimit(user, wsi, 26, 1, 30, nums(5, 7, 15, 17));
-		
-		wsi = new WorkspaceIdentifier("skiplimit3");
-		ws.createWorkspace(user, wsi.getName(), false, null, null).getId();
+		checkObjectLimit(user, wsi2, 5, 1, 6, nums(5));
+		checkObjectLimit(user, wsi2, 6, 1, 8, nums(5, 7));
+		checkObjectLimit(user, wsi2, 12, 1, 14, nums(5, 7));
+		checkObjectLimit(user, wsi2, 13, 1, 16, nums(5, 7, 15));
+		checkObjectLimit(user, wsi2, 14, 1, 18, nums(5, 7, 15, 17));
+		checkObjectLimit(user, wsi2, 14, 1, 18, nums(5, 7, 15, 17));
+		checkObjectLimit(user, wsi2, 15, 1, 19, nums(5, 7, 15, 17));
+		checkObjectLimit(user, wsi2, 16, 1, 20, nums(5, 7, 15, 17));
+		checkObjectLimit(user, wsi2, 17, 1, 21, nums(5, 7, 15, 17));
+		checkObjectLimit(user, wsi2, 18, 1, 22, nums(5, 7, 15, 17));
+		checkObjectLimit(user, wsi2, 26, 1, 30, nums(5, 7, 15, 17));
+		
+		final WorkspaceIdentifier wsi3 = new WorkspaceIdentifier("skiplimit3");
+		ws.createWorkspace(user, wsi3.getName(), false, null, null).getId();
 		objs.clear();
 		for (int i = 1; i < 251; i++) {
 			objs.add(new WorkspaceSaveObject(getRandomName(), new HashMap<String, String>(),
 					SAFE_TYPE1, null, new Provenance(user), false));
 		}
-		ws.saveObjects(user, wsi, objs, getIdFactory(0));
+		ws.saveObjects(user, wsi3, objs, getIdFactory(0));
 		
-		for (int i = 1; i < 251; i++) {
-			List<ObjectIdentifier> oi =
-					Arrays.asList(new ObjectIdentifier(wsi, i));
+		for (long i = 1; i < 251; i++) {
+			final List<ObjectIdentifier> oi = list(
+					ObjectIdentifier.getBuilder(wsi3).withID(i).build());
```

Comment:
Figuring out I was still using the builder from what's now called wsi1 took a long time when this test started failing. This is why final variables (and shorter tests that don't reuse variable names) are Very Good Things